### ISSUE TITLE
TO API allows creating a DS on the "ALL" CDN

### ISSUE THREAD

**Issue by rawlinp** [apache/trafficcontrol#4324]
## I'm submitting a ...

- [x] bug report
- [ ] new feature / enhancement request
- [ ] improvement request (usability, performance, tech debt, etc.)
- [ ] other 

## Traffic Control components affected ...

- [ ] CDN in a Box
- [ ] Documentation
- [ ] Grove
- [ ] Traffic Control Client
- [ ] Traffic Monitor
- [x] Traffic Ops
- [ ] Traffic Ops ORT
- [ ] Traffic Portal
- [ ] Traffic Router
- [ ] Traffic Stats
- [ ] Traffic Vault
- [ ] unknown

## Current behavior:

The Traffic Ops API allows creating/updating a delivery service to be on the "ALL" CDN, which is invalid and won't work.

## Expected / new behavior:

The TO API should prohibit assigning Delivery Services to the "ALL" CDN.

## Minimal reproduction of the problem with instructions:

Using the TO API, create a Delivery Service with the `cdnId` of the "ALL" CDN.

**Comment by mitchell852** [comment]
Does this fall under the category of "don't do that"? ;)

**Comment by rawlinp** [comment]
Yeah, hence the minor label. The DS obviously wouldn't work, but it was kind of confusing when I realized I used the wrong `cdnId` in my request and ended up with the "ALL" CDN.
